spec/lib/tasks/existing_firms_sign_up_task_spec.rb

```diff
@@ -1,107 +1,117 @@
+class StubInvitationHelper
+  def invitation_url(user)
+    "invitation for user #{user.id}"
+  end
+end
+
 module Tasks
   RSpec.describe ExistingFirmsSignUpTask do
     describe '#notify' do
-      let!(:bills_firm) { create :firm }
-      let!(:bill) { create :principal, email_address: 'bill@example.com', firm: bills_firm }
+      let!(:bills_firm) { create :firm, registered_name: 'Wright, Johnston & MacKenzie LLP' }
+      let!(:bill) do
+        principal = build :principal
+        principal.email_address =  'bill@example.com'
+        principal.firm = bills_firm
+        principal.first_name = 'Bill'
+        principal.last_name = 'Junior, the third'
+        principal.save!
+        principal
+      end
```

Comment:
For me, when I read it, `bill` gives less context as I'm not sure what kind of a thing `bill` is. Since we're only using one principal here(?) `principal` seems like an adequate name to me.